### ISSUE TITLE
Add nestedBuilder capability on specific fields (BuiltValueField)

### ISSUE THREAD

**Issue by fromlabs** [google/built_value.dart#918]
I have a big project using `built_value` and I'm very happy.
Over time my model grown up with inheritance and generics and I've found a problem with nested builders maybe related to #426 .
Using `@BuiltValue(nestedBuilders: false)` as a workaround works and it's fine but this refactoring is quite challenging because I've classes with a lot of attributes (built values) and it would be better to disable builders on specific fields.
Having `@BuiltValueField` annotation with `nestedBuilder` and `autoCreateNestedBuilder` attributes could help me a lot.

**Comment by davidmorgan** [comment]
Makes sense as a feature request--but unlikely to happen soon I'm afraid.

**Comment by fromlabs** [comment]
Can I help you with a pull request?

**Comment by davidmorgan** [comment]
That's likely to be hard right now: the library is migrated to null safety, but the current SDK release hides null safety behind an experiment flag. So, it will be hard to make changes, and we anyway won't be able to publish it until an SDK is released with null safety enabled by default.

When that happens--I think the goal is, later this year--a PR would help, yes :)

**Comment by alanrussian** [comment]
+1 I ran into this too. David, would it be possible to review this pull request soon?

My use case is that I want to create a new BuiltValue class X and add it to a big and commonly used BuiltValue class Y. I don't want anyone to be able to set X in Y unless they are allow listed. I put X in a separate file and was hoping to use bazel visibilities. However, I realized developers could circumvent this build visibility restriction by using a nested builder.

```dart
// No import on Y is required here with nested builders. 
final x = X((xBuilder) => xBuilder.yBuilder.yField = true);
```
